server/services/recommendedVersionsService.ts

```diff
@@ -0,0 +1,208 @@
+      )
+
+      // Legacy helm shape in versions
+      if (!helmGenericPrometheusAlerts || !helmGenericService) {
+        const helm = versions.helm as Record<string, unknown>
+        const deps = (helm?.dependencies as Record<string, unknown>) || {}
+        helmGenericPrometheusAlerts =
+          helmGenericPrometheusAlerts || this.parseVersionValue(deps['generic-prometheus-alerts'])
+        helmGenericService = helmGenericService || this.parseVersionValue(deps['generic-service'])
+      }
+
+      // Fallback to values container
+      if (!helmGenericPrometheusAlerts || !helmGenericService || !gradleHmppsGradleSpringBoot) {
+        const vHelmDeps = (values.helm_dependencies as Record<string, unknown>) || {}
+        const vGradle = (values.gradle as Record<string, unknown>) || {}
+        helmGenericPrometheusAlerts =
+          helmGenericPrometheusAlerts || this.parseVersionValue(vHelmDeps.generic_prometheus_alerts)
+        helmGenericService = helmGenericService || this.parseVersionValue(vHelmDeps.generic_service)
+        // Fallback to hyphenated keys under values.helm_dependencies
+        if (!helmGenericPrometheusAlerts) {
+          helmGenericPrometheusAlerts = this.parseVersionValue(vHelmDeps['generic-prometheus-alerts'])
+        }
+        if (!helmGenericService) {
+          helmGenericService = this.parseVersionValue(vHelmDeps['generic-service'])
+        }
+        gradleHmppsGradleSpringBoot =
+          gradleHmppsGradleSpringBoot || this.parseVersionValue(vGradle.hmpps_gradle_spring_boot)
+      }
+
+      // Legacy helm shape in values
+      if (!helmGenericPrometheusAlerts || !helmGenericService) {
+        const vHelm = (values.helm as Record<string, unknown>) || {}
+        const vDeps = (vHelm.dependencies as Record<string, unknown>) || {}
+        helmGenericPrometheusAlerts =
+          helmGenericPrometheusAlerts || this.parseVersionValue(vDeps['generic-prometheus-alerts'])
+        helmGenericService = helmGenericService || this.parseVersionValue(vDeps['generic-service'])
+      }
+
+      const versionsKeys = Object.keys(versions || {})
+      const valuesKeys = Object.keys(values || {})
+      logger.info(
+        `[RecommendedVersions] Strapi containers: versions keys=${JSON.stringify(versionsKeys)}, values keys=${JSON.stringify(
+          valuesKeys,
+        )}`,
+      )
+      logger.info(
+        `[RecommendedVersions] Strapi extracted helm(gpa=${helmGenericPrometheusAlerts || 'missing'}, gs=${
+          helmGenericService || 'missing'
+        }), gradle(hgsb=${gradleHmppsGradleSpringBoot || 'missing'})`,
+      )
+
+      if (helmGenericPrometheusAlerts || helmGenericService || gradleHmppsGradleSpringBoot) {
+        return {
+          helm_generic_prometheus_alerts: helmGenericPrometheusAlerts,
+          helm_generic_service: helmGenericService,
+          gradle_hmpps_gradle_spring_boot: gradleHmppsGradleSpringBoot,
+        }
+      }
+      return null
```

Comment:
This should be outside the try/catch and also removed from the catch as the final return in the try and catch are both the same.